docs/user/key_concepts_overview.md

```diff
@@ -0,0 +1,335 @@
+
+```
+@dataclass
+class Maker(MSONable):
+    """
+    Base maker (factory) class for constructing :obj:`Job` and :obj:`Flow` objects.
+    [...]
+    """
+    def make(...) -> jobflow.Flow | jobflow.Job:
+        """Make a job or a flow - must be overridden with a concrete implementation."""
+        raise NotImplementedError
+    [...]
+    def update_kwargs(...):
+```
+
+When implementing a new job or flow maker, it's crucial to take care of the functions that raise an `NotImplementedError`. Functions like the `make` function have to be overridden for each specific job or flow maker with its own specific functionalities.
+
+### Examples
+
+Let's have a closer look at the `LobsterMaker` as an example for a `Job Maker`:
+
+```
+@dataclass
+class LobsterMaker(Maker):
+    """
+    LOBSTER job maker.
+    [...]
+    """
+    name: str = "lobster"
+    [...]
+
+    @job(output_schema=LobsterTaskDocument, data=[CompleteCohp, LobsterCompleteDos])
+    def make(
+        self,
+        wavefunction_dir: str | Path = None,
+        basis_dict: dict | None = None,
+    ) -> LobsterTaskDocument:
+        """
+        Run a LOBSTER calculation.
+        [...]
+        """
+```
+This class incorporates [LOBSTER](http://cohp.de/) specific input and output data, i.e. the `wavefunction_dir` and `basis_dict` as input in `make` that returns the `LobsterMaker`-class specific output as a `TaskDocument`. Also, note how the `make` functions is annotated by the `@job` decorator. This small addition to `make` turns the maker into a job maker.
+As the name "job maker" implies, this maker will then create jobs to execute the LOBSTER runs and store the output in the `LobsterTaskDocument` format.
+
+What changes need to be made to turn a maker into a `Flow Maker`? In that case, the `make` function needs to be adjusted to return a `Flow` object instead of a task document.
+As an example we take the `BasePhononMaker` (that can be used in combination with VASP or the various (machine-learned) force fields):
+```
+@dataclass
+class BasePhononMaker(Maker, ABC):
+    """
+    Maker to calculate harmonic phonons with a DFT/force field code and Phonopy.
+
+    [...]
+    """
+
+    name: str = "phonon"
+    [...]
+
+    def make(...) -> Flow:
+```
+This maker will return a flow that provides all the necessary steps and subroutines that are needed to complete the phonon calculations.
+Oftentimes, such flows then involve dynamic jobs. For the phonon calculations this means that the number of supercells with individual atomic displacements will be decided upon runtime.
+In this particular case, the flow maker `BasePhononMaker` is also inheriting from `ABC` (Abstract Base Classes). <!--more comprehensive explanation on ABCs?-->
```

Comment:
```suggestion
In this particular case, the flow maker `BasePhononMaker` is also inheriting from `ABC` (Abstract Base Classes). <!--more comprehensive explanation on ABCs?-->
```
I am actually not even sure the abstract class is still needed as we unified the previous dict names, if I am not mistaken. Likely, https://github.com/materialsproject/atomate2/blob/974b9cddff8edcd1773631283962dd88f163a3ad/src/atomate2/common/flows/phonons.py#L363 could be even removed. Thus, I wouldn't add an explanation on this here.

Reply:
okay, taking back that the last part could be removed. I have just seen one of the newer implementations with yet another way to connect jobs.